### ISSUE TITLE
[Feature Request] Modularize the MMORPG Kit

### ISSUE THREAD

**Issue by Eniotnacram** [suriyun-production/mmorpg-kit-docs#2542]
Hello,

As dedicated and long-time users of the MMORPG Kit, many of us believe that a more modular approach would greatly enhance the flexibility and usability of the kit. We understand that this is a significant undertaking, but we feel that gradually transitioning towards this model could be highly beneficial.

Instead of adding every new feature directly into the core, we suggest developing optional features like ladder logic, morphing, building (construction), guild wars, etc. as separate add-ons. This would keep the core lean and allow users to integrate only the features that suit their project needs.

We know this would require time and effort, but perhaps it can be done progressively, bit by bit. Even small steps in this direction would be appreciated by the community and help ensure the kit remains adaptable and performant as more features are introduced.

Thank you very much

**Comment by Eniotnacram** [comment]
On a side note, you could even make those new add-ons (features) paid and sell them on the Unity asset store. So your work and time on those would be more beneficial for you. I can't even grasp the fact that the kit is only 80$ still, with all of those features added and time dedicated to them.

**Comment by insthync** [comment]
Yes, but most thing really can't without core changes.

**Comment by insthync** [comment]
Some are already been separated projects (guildwar, login with facebook/google, auction house and so on). But some features which required CORE changes are mostly relates to entity movement/character model's animation, those are features that were requested frequently (new animations, new option for animations, dash/ladder requires new animations, dash/ladder requires new movement states).

There are very few of new features those were not require movement/animation changing such as player housing, morphing, advanced pet but there are new requests for movement/animation changing every month, such as dash, after the dash done then you asking for dash extension. Yesterday I saw a new question about animation again. I fixed many of root-motion movement, and I might have to fix it after ladder climbing is done, and might have to add an extension for ladder climbing, and may have new request for free climb, may have stamina requirement for climbing and so on.

I closed a post that request for a attacking force feature which will push enemy backward because it has no one pay for dash yet, so in the future I won't develop a new paid features without paid, everyone who request for paid request must pay before I start or I will sell it as addons on asset store.

Some paid request features were wanted by many users but always has only one guy pay for it.

So I would said I already try to make it lean, but I didn't try to sell any as addons in asset store yet, so next time I will sell some as addons on asset store (not include previous projects on github). I also hope it might get rid of hesitate feeling to develop paid request features.
